lib/dotcom/utils/date_time.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Dotcom.Utils.DateTime do
+  @moduledoc """
+  A collection of functions for working with date_times.
+
+  Consuming modules are responsible for parsing or converting date_times.
+  They should *always* call `coerce_ambiguous_time/1` before using a date_time.
+  This is mainly because Timex has so many functions that serve as entry points to date_times.
+  Those functions can return ambiguous date_times during DST transitions.
+  """
+
+  require Logger
+
+  use Timex
+
+  @typedoc """
+  A range of time that can be open ended on either side.
+  """
+  @type date_time_range() ::
+          {DateTime.t(), DateTime.t()} | {nil, DateTime.t()} | {DateTime.t(), nil}
+
+  @timezone Application.compile_env!(:dotcom, :timezone)
+
+  @doc """
+  Returns the timezone for the application.
+  """
+  def timezone(), do: @timezone
+
+  @doc """
+  Get the date_time in the set @timezone.
+  """
+  @spec now() :: DateTime.t()
+  def now(), do: Timex.now(@timezone)
+
+  @doc """
+  In the default case, we'll return a DateTime when given one.
+
+  Timex can give us ambiguous times when we "fall-back" in DST transitions.
+  That is because the same hour occurs twice.
+  In that case, we choose the later time.
+
+  Timex will return an error if the time occurs when we "spring-forward" in DST transitions.
+  That is because one hour a day does not occur--2am to 3am.
+  In that case, we set the time to 3am.
+
+  If we are given something thatis not a DateTime, AmbiguousDateTime, or an error tuple, we log the input and return `now`.
+  """
+  @spec coerce_ambiguous_time(DateTime.t() | Timex.AmbiguousDateTime.t() | {:error, term()}) ::
+          DateTime.t()
+  def coerce_ambiguous_time(%DateTime{} = date_time), do: date_time
```

Comment:
There're lots of spots in Dotcom working with `%NaiveDateTime{}`... we probably want to make this work with those too.

Reply:
The idea behind the library is that the caller is responsible for handling that. One of the reasons that the old DateTime code is such a mess is because it was trying to do too much in the sense of trying to accept everything that client code could throw at it. The only reason we have this function is because Timex can return these things and we need to make default choices about how to handle them.

Reply:
I.e., I believe we have so much handling of naive date times because of integrations with front-end code. Rather than saying...sure, just give us a naive date time, we make that code give us a date time. Otherwise, every function we write now has to handle naive date times and try to do the conversion.

Reply:
Okay, fair enough!